chapters/statements.tex

```diff
@@ -177,6 +177,10 @@ \subsection{For-Statement}\label{for-statement}
 \end{lstlisting}
 The rules for \lstinline!for!-statements are the same as for \lstinline!for!-expressions in \cref{explicit-iteration-ranges-of-for-equations} -- except that the \lstinline!expression! of a \lstinline!for!-statement is not restricted to a parameter-expression.
 
+If the \lstinline!for!-statement contains expressions that will generate events, the range shall be an evaluable expression.
```

Comment:
Use standard formulation _event-generating expression_, and deal with possibility for multiple expressions in the range:
```suggestion
If the \lstinline!for!-statement contains event-generating expressions, any expression in \lstinline[language=grammar]!for-index! shall be evaluable.
```